src/bin/rustup-init.rs

```diff
@@ -26,29 +26,36 @@ use rustup::cli::rustup_mode;
 #[cfg(windows)]
 use rustup::cli::self_update;
 use rustup::cli::setup_mode;
-use rustup::currentprocess::{process, varsource::VarSource, with, OSProcess};
+use rustup::currentprocess::{process, varsource::VarSource, with_runtime, OSProcess};
 use rustup::env_var::RUST_RECURSION_COUNT_MAX;
 use rustup::is_proxyable_tools;
 use rustup::utils::utils;
+use tokio::runtime::Builder;
 
 fn main() {
     #[cfg(windows)]
     pre_rustup_main_init();
 
-    let process = OSProcess::default();
-    with(process.into(), || match maybe_trace_rustup() {
-        Err(e) => {
-            common::report_error(&e);
-            std::process::exit(1);
+    let process = OSProcess::new();
```

Comment:
Why the change from `OSProcess::default()` to `OSProcess::new()` here? That seems unrelated.

Reply:
It was the elevation of stderr_is_a_ttry into the OSProcess struct, that meant that either a manual Default implementation was needed, or new(), and I have no idea why I decided to do both :).

I made stderr_is_a_tty state to resurrect a previous efficiency optimisation we had for Windows, so that tty probing only happens one per process, not once per terminal construction. It really should be a separate patch; and will need redoing with your other refactoring.

Reply:
I'm not sure I understand what you mean here. Seems that `stderr_is_a_tty` is already state in `OSProcess` on `master`, and `default()` and `new()` are interchangeable, so I'd prefer to remove this change from this PR.